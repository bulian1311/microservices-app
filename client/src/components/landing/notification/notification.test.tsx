import React from "react";
import { render, RenderResult } from "@testing-library/react";
import Notification from "./notification.component";

let documentBody: RenderResult;

beforeEach(() => {
  documentBody = render(
    <Notification
      timer={200}
      notification={{ id: "1", title: "Test", img: "" }}
    />
  );
});

test("Notification component match snapshot.", () => {
  const { baseElement } = documentBody;
  expect(baseElement).toMatchSnapshot();
});

test("Notification component unmount.", () => {
  documentBody.getByTestId("notification");

  setTimeout(() => {
    expect(documentBody.queryByTestId("notification")).toBeNull();
  }, 300);
});
