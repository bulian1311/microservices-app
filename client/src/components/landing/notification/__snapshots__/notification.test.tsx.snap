// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification component match snapshot. 1`] = `
<body>
  <div>
    <div
      class="sc-gsTCUz drnWZo"
      data-testid="notification"
    >
      <div
        class="sc-hKgILt Yjdfg"
      >
        <div
          class="sc-bdfBwQ fdDwxF"
        >
          <img
            class="sc-dlfnbm jSKCcd"
            src=""
          />
        </div>
      </div>
      <span
        class="sc-eCssSg oRpWe"
      >
        Test
         добавлен в корзину
      </span>
    </div>
  </div>
</body>
`;
